src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -280,6 +291,36 @@ protected function getArrayableRelations()
         return $this->getArrayableItems($this->relations);
     }
 
+    /**
+     * Determine whether an attribute will be encrypted.
+     *
+     * @return bool
+     */
+    protected function hasEncryption($key)
+    {
+        return in_array($key, $this->encrypt);
+    }
+
+    /**
+     * Encrypt attribute's value.
+     *
+     * @return void
+     */
+    public function encryptAttribute($key, $value)
+    {
+        $this->attributes[$key] = is_null($value) ? null : encrypt($value);
```

Comment:
@JustCarty This will break when the illuminate/database package is used outside of a full Laravel framework application.

Reply:
Isn't this the actual problem why previous attempts were not endorsed?

- This would mean that illuminate/database would need a dependency to illuminate/encryption
- you can't use `encrypt()` which calls `app()` which you also can't use here; you would need to directly access the container

Don't forget this package needs to work without Laravel, too.

Maybe a package "on top" would be better suited, say `illuminate/database-encryption` which depends on `illuminate/database` and `illuminate/encryption` and provides some trait or other means to spice up selected models; then illuminate/encryption would only need to provide some hook system to "make" it work, if the current infrastructure isn't enough.

Note: no idea about the feasibility, this was just a quick thought coming to my mind.

Reply:
I think it's fair to put it on composer suggest and throw an exception saying you should install illuminate/encryption to use this feature. Optional dependencies exist in a lot of places. Migrations has one with doctrine, for instance.

Reply:
I don't think this should be merged to Eloquent in any shape or form as this is totally out of scope of what an ORM should be responsible for. Eloquent has enough responsibilities with the accessor and mutator "magic" as is already IMHO.

If this really needs to be merged though I'd implement it the way @GrahamCampbell suggested, by adding a static property to the `Model` class for the encrypter and setting it in the framework booting process and then to check in these methods if it isn't set to do nothing (or throw an exception).